### DISCUSSION TITLE
一重启占用直接拉满

### DISCUSSION THREAD

**Discussion by ghost** [Arondight/Adachi-BOT#783] [Q&A]
不知道是不是我的问题，添加Cookie后重启过会占用直接拉满，先停止再启动也一样，只能通过重启服务器占用就会恢复正常，是chrome的原因吗

**Comment by ghost** [comment]
忘记说了，配置CentOS7 2h4g 重启后占用直接70%（cpu 内存两者都是）往上，重启服务器恢复正常10+

**Reply by mark9804** [reply]
是 CPU 和内存占用都占了这么多，还是只有 CPU ？我看下面的图是只有 CPU 的

**Comment by mark9804** [comment]
是啥占用拉满了，然后跑一下 `npm run tool-info`

**Reply by ghost** [reply]
> 是啥占用拉满了，然后跑一下 `npm run tool-info`

CPU和内存，都70+%，只有通过重启服务器才能解决

**Reply by mark9804** [reply]
啊不是，我没说清楚，我的意思是具体什么进程占用了这么多计算资源

**Reply by ghost** [reply]
> 啊不是，我没说清楚，我的意思是具体什么进程占用了这么多计算资源
![QQ截图20220405090237](https://user-images.githubusercontent.com/77124366/161657892-890df314-c1b7-4a52-964f-d8c98d912012.png)
![QQ截图20220405090321](https://user-images.githubusercontent.com/77124366/161657894-6cc4654d-04f8-4a3d-8490-9418581804f8.png)
应该说只是CPU占用变高，重启就恢复，
![QQ截图20220405090553](https://user-images.githubusercontent.com/77124366/161658040-aa0805b6-2971-4893-8391-9350374be757.png)

**Reply by mark9804** [reply]
麻烦跑一下 `npm run tool-status` ，或者直接跟机器人发送 `系统信息` ，我们看一下具体的软硬件配置，然后启动命令我看到你使用的是 `node app.js` ，没有用我们提供的 `npm run init` 或者是 `npm run start` 对吗

@Arondight 这里有一个主程序跑出了太高 CPU 占用的问题

**Reply by ghost** [reply]
> 
![QQ截图20220405091547](https://user-images.githubusercontent.com/77124366/161658898-402fade9-184a-49ef-a69c-a2537bcee935.png)
这是重启服务器后的截图
我重启机器人确实用了npm run restart或者停止启动npm run stop/npm run start这个命令的，至于出现node app.js我也想不出（小白一枚）

**Comment by Arondight** [comment]
日志中没得报错？

**Reply by ghost** [reply]
> 日志中没得报错？

木有报错，机器人也是正常使用，只是重启占用就高，使用起来一切正常，没啥不能用的

**Reply by mark9804** [reply]
需要发图的功能也是正常的吗，试试看 `今日素材` 或者 `信息绫人`

**Reply by Arondight** [reply]
截图一下 top 命令看看，你那个图我不太会看……看一下 cpu 总体的状态 us 和 id 多少

**Reply by Arondight** [reply]
另外确认一下

1. 只有加 cookie 才会表现出占用高？
2. 加了几个 cookie

**Reply by ghost** [reply]
> 

刚刚使用，正常的，help命令里所有指令都是能正常使用

**Reply by ghost** [reply]
> top

Cookie加了十个左右，我这里的情况是重启机器人就会占用高，然后重启服务器就会恢复正常，不管前置动作有没有加Cookie，都是这样

**Reply by ghost** [reply]
> top
![QQ截图20220405094320](https://user-images.githubusercontent.com/77124366/161662797-581c1471-b63b-4164-b4b7-663acea57cf6.png)
这是占用正常的情况

**Reply by mark9804** [reply]
只要重启机器人就会100%复现 CPU 占用高的情况吗，我们想看看不正常的 top

**Reply by ghost** [reply]
> 截图一下 top 命令看看，你那个图我不太会看……看一下 cpu 总体的状态 us 和 id 多少
![QQ截图20220405094734](https://user-images.githubusercontent.com/77124366/161663905-6bc24de5-6cdb-44e9-8b00-88950270cab1.png)
![QQ截图20220405095407](https://user-images.githubusercontent.com/77124366/161663911-67e2f0d9-82f2-4c7b-aebf-d861e9e0f722.png)
好家伙，难不成宝塔面板的锅？理论来说没关系啊😂

**Reply by mark9804** [reply]
没有头绪，不过看起来确实不是这个项目的 bug…

**Reply by ghost** [reply]
> 没有头绪，不过看起来确实不是这个项目的 bug…

嗯好，我知道了，top也看出占用不是很高，应该我这边的原因，辛苦了~

**Reply by Arondight** [reply]
你理解错了，你这图确实 cpu 占用率很高，而且是单纯的高，不是 io 堵塞了导致，你那个不正常的图，停掉 bot ，看看 us 是不是立刻下去了

**Reply by Arondight** [reply]
至于原因不好说，有可能是浏览器在空转，也有可能是你的 vps 不行例如 cpu 超分严重，这个还真不好查，你可以

1. 升级浏览器版本试试，现在 100 出来了，上周公司里的 archlinux 已经收到更新了
2. 确认一下 vps 是不是有啥超分问题，例如几十块钱一年的轻应用大礼包

**Reply by ghost** [reply]
> 你理解错了，你这图确实 cpu 占用率很高，而且是单纯的高，不是 io 堵塞了导致，你那个不正常的图，停掉 bot ，看看 us 是不是立刻下去了
![QQ截图20220405101026](https://user-images.githubusercontent.com/77124366/161665641-988d5658-a878-4d38-a66b-b03492514d03.png)
![QQ截图20220405101137](https://user-images.githubusercontent.com/77124366/161665646-af408e39-386a-486d-aa5a-ac54668ad687.png)
![QQ截图20220405101230](https://user-images.githubusercontent.com/77124366/161665652-e41b1efe-bc5f-4960-b240-72bafb6cfec2.png)
停了确实正常，一开就拉满，超分问题的话，我用的腾讯轻量云，有没有我倒是不知

**Reply by Arondight** [reply]
先用着看看吧，可以的话贴一下 git log -1，一是我们这儿复现不了，二是这类问题也不好查

**Reply by ghost** [reply]
> git log -1

commit eee7e5b8dbbd803572e36c9d260b32c81329ba63
Author: Qin Fandong <shell_way@foxmail.com>
Date:   Mon Apr 4 21:04:59 2022 +0800

    Fix README


没事，只要不重启就没事，顶多加个Cookie重启下服务器而已😂也不经常加，所以不用频繁重启的

**Reply by mark9804** [reply]
如果是 chromium 不正常的话占用高的进程名应该是我们熟悉的 n 进宫 chromium ，但是现在宝塔显示是主程序占用高，log 也没有异常，我就没啥头绪

**Reply by Arondight** [reply]
我下午本地起个虚拟机查一下，主要也是很难想象啥事不干的 app.js 产生占用

**Reply by ghost** [reply]
> 我下午本地起个虚拟机查一下，主要也是很难想象啥事不干的 app.js 产生占用

okk辛苦了，我宝塔里面搭建的网站用的是nginx+php，没有用到nodejs的，唯一用到的是本Bot，

**Reply by Arondight** [reply]
实在复现不出来，包括在腾讯轻量云上，我暂时记一下这问题吧，目前也做不了什么
